cdap-examples/SpamClassifier/src/test/java/co/cask/cdap/examples/sparkstreaming/SpamClassifierTest.java

```diff
@@ -0,0 +1,178 @@
+import co.cask.common.http.HttpRequests;
+import co.cask.common.http.HttpResponse;
+import com.google.common.base.Charsets;
+import com.google.common.io.Closeables;
+import org.apache.twill.internal.kafka.EmbeddedKafkaServer;
+import org.apache.twill.internal.kafka.client.ZKKafkaClientService;
+import org.apache.twill.internal.zookeeper.InMemoryZKServer;
+import org.apache.twill.kafka.client.Compression;
+import org.apache.twill.kafka.client.KafkaClientService;
+import org.apache.twill.kafka.client.KafkaPublisher;
+import org.apache.twill.zookeeper.ZKClientService;
+import org.junit.AfterClass;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Properties;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Test for {@link SpamClassifier}
+ */
+public class SpamClassifierTest extends TestBase {
+
+  private static final String KAFKA_TOPIC = "someTopic";
+  private static final String KAFKA_BROKER_ID = "1";
+  private static ZKClientService zkClient;
+  private static KafkaClientService kafkaClient;
+
+  private static InMemoryZKServer zkServer;
+  private static EmbeddedKafkaServer kafkaServer;
+  private static int kafkaPort;
+
+  @BeforeClass
+  public static void initialize() throws Exception {
+    TestBase.initialize();
+    zkServer = InMemoryZKServer.builder().setDataDir(TMP_FOLDER.newFolder()).build();
+    zkServer.startAndWait();
+
+    kafkaPort = Networks.getRandomPort();
+    kafkaServer = new EmbeddedKafkaServer(generateKafkaConfig(zkServer.getConnectionStr(), kafkaPort,
+                                                              TMP_FOLDER.newFolder()));
+    kafkaServer.startAndWait();
+
+    zkClient = ZKClientService.Builder.of(zkServer.getConnectionStr()).build();
+    zkClient.startAndWait();
+
+    kafkaClient = new ZKKafkaClientService(zkClient);
+    kafkaClient.startAndWait();
+  }
+
+  @AfterClass
+  public static void cleanup() {
+    kafkaClient.stopAndWait();
+    zkClient.stopAndWait();
+    kafkaServer.stopAndWait();
+    zkServer.stopAndWait();
+  }
+
+  @Test
+  public void test() throws Exception {
+
+    // Deploy the KafkaIngestionApp application
+    ApplicationManager appManager = deployApplication(SpamClassifier.class);
+
+    ingestTrainingData();
+
+    publishKafkaMessages();
+
+    // start spark streaming program
+    SparkManager sparkManager = appManager.getSparkManager(SpamClassifierProgram.class.getSimpleName());
+    Map<String, String> runtimeArgs = new HashMap<>();
+    runtimeArgs.put("kafka.brokers", "127.0.0.1:" + kafkaPort);
+    runtimeArgs.put("kafka.topics", KAFKA_TOPIC);
+    sparkManager.start(runtimeArgs);
+    // wait for sometime for processing
+    TimeUnit.SECONDS.sleep(15);
```

Comment:
Since you'll be expected two messages being written to the dataset, I would say do a `Tasks.waitFor` and fetch the message spam info inside.